platform/x11/detect.py

```diff
@@ -122,25 +123,21 @@ def configure(env):
     # Cross-compilation
     # TODO: Support cross-compilation on architectures other than x86.
     host_is_64_bit = sys.maxsize > 2**32
+    is_x86_target = "x86" in env["arch"] if env["arch"] else "x86" in detect_arch()
+
+    if env["arch"] == "" and not is_x86_target:
+        env["arch"] = detect_arch()
```

Comment:
If `env["arch"]` is empty, and the engine is compiled with `bits=32` on an arm64 host, wouldn't this cause `env["arch"]` to be set to arm64, mismatching `env["arch"]` with `env["bits"]`?

Reply:
Ah, good point, maybe we could do something like:

```python
    host_is_64_bit = sys.maxsize > 2**32
    is_x86_target = "x86" in env["arch"] if env["arch"] else "x86" in detect_arch()

    if env["arch"] == "" and not is_x86_target:
        env["arch"] = detect_arch()
        if env["bits"] != "default":
                env["arch"] = env["arch"].replace("32", env["bits"]).replace("64", env["bits"])
```

But feels very hacky...
Or I could just revert that change and just never force the `env["arch"]` beside what's already done below for rv64/ppc64/ppc.

Reply:
That code would allow for `bits=32` on rv64 host setting the arch to rv32 which we don't support.

Would it be acceptable to require users to be explicit about the architecture? I am not sure if this would break any existing workflows, based on your PR title I guess non-x86 cross-compilation was not working before.

```py
    if env["arch"] == "" and env["bits"] != "default" and not is_x86_target:
        print("You must specify the architecture. For example, `bits=" + env["bits"] + "arch=" + detect_arch().replace("32", env["bits"]).replace("64", env["bits"]) + "`")
        Exit(255)
```

Admittedly this is basically the same logic as what you put, but putting more of the burden on users. It would still print `rv32` when using bits=32 on rv64 but at least it doesn't select that invalid config automatically. So anyway it's not really much better than what you suggest. Either of these are probably fine.